apsuite/commissioning_scripts/measure_id_integrals.py

```diff
@@ -0,0 +1,206 @@
+"""."""
+from threading import Thread as _Thread, Event as _Event
+import time as _time
+import numpy as _np
+from siriuspy.devices import SOFB, APU, Tune
+
+from epics import PV
+from pymodels import si
+from .base import BaseClass
+
+
+class IDParams:
+    """."""
+
+    def __init__(self, phases=None, meas_type='static'):
+        """."""
+        self.phases = phases
+        self.meas_type = meas_type
+        if self.meas_type == 'static':
+            self.phase_speed = 0.5
+            self.sofb_mode = 'SlowOrb'
+            self.sofb_buffer = 20
+            self.wait_sofb = 1
+            self.wait_to_move = 0
+        elif self.meas_type == 'dynamic':
+            self.phase_speed = 0.5
+            self.sofb_mode = 'Monit1'
+            self.sofb_buffer = 1
+            self.wait_sofb = 10
+            self.wait_to_move = 1
+
+    def __str__(self):
+        """."""
+        ftmp = '{0:26s} = {1:9.6f}  {2:s}\n'.format
+        stmp = '{0:35s}: {1:}  {2:s}\n'.format
+        dtmp = '{0:26s} = {1:9d}  {2:s}\n'.format
+        stg = ftmp('phase_speed', self.phase_speed, '')
+        stg += stmp('sofb_mode', self.sofb_mode, '')
+        stg += dtmp('sofb_buffer', self.sofb_buffer, '')
+        return stg
+
+
+class MeasIDIntegral(BaseClass):
+    """."""
+
+    def __init__(self,
+                 model,
+                 id_name=None, phases=None, meas_type='static'):
+        """."""
+        super().__init__()
+        self.model = model
+        self.famdata = si.get_family_data(model)
+        self.params = IDParams(phases, meas_type)
+        self.id_name = id_name
+        self.devices['apu'] = APU(APU.DEVICES.APU22_09SA)
```

Comment:
This class is not specific to this ID, right? maybe implement other IDs in `APU` and use `self.id_name` as argument?

Reply:
yes, perfect, thanks. 

Some parts of the code are incomplete or will be changed. I just created this draft PR to get the feedback from you while I'm working on the code and also to avoid pushing a huge amount of information to be reviewed at once.